iris-mpc-cpu/src/execution/hawk_main/matching.rs

```diff
@@ -93,17 +98,31 @@ impl Step1 {
                 Vec::with_capacity(capacity / 2),
                 Vec::with_capacity(capacity / 2),
             ],
+            luc_ids: Vec::new(),
         }
     }
 
     fn missing_vector_ids(&self, side: usize) -> VecEdges<VectorId> {
         let other_side = 1 - side;
-        self.anti_join[other_side].clone()
+        let anti_join = &self.anti_join[other_side];
+
+        chain!(anti_join, &self.luc_ids).cloned().collect_vec()
```

Comment:
This could use a `unique()` to reduce work a bit in some scenarios, although the correctness should still work without it, right?

Reply:
Good point, I added that.

This required aligning to a single kind of "versioned" ID. Then I could also remove the suspicious idea of ANY_VERSION too.